### PR TITLE
adding keyboard support and styling

### PR DESCRIPTION
**Pull request by christulin**
-Add onKeyDown to button element of Square component
-Add new state of 'test' to handle classname of each square component (can update name after review)
-Add applicable CSS for active squares (we can update later)
-Add handleKeyDown() function to Gameboard.js to update
-autofocus to true on first Square component rendered

Right now keyboard support only handles left and right.  V2 can be to add up/down functionality (will require some basic math)

### CODE REVIEW

File: src/Components/GameBoard.js [hunk 1]
```diff
@@ -1,20 +1,64 @@
 import React from 'react';
-import { Link } from 'react-router-dom';
```
Comment (christulin) [inline src/Components/GameBoard.js:2]: module not being used

File: src/Components/GameBoard.js [hunk 2]
```diff
@@ -1,20 +1,64 @@
...
   constructor(props) {
     super(props);
-
+    this.handleKeyDown = this.handleKeyDown.bind(this)
     this.state = {
       squareArray: Array(9).fill(null),
       history: [],
       xIsNext: true,
+      squareActive: ['square active', 'square', 'square', 'square', 'square', 'square', 'square', 'square', 'square'],
+      cursor: [0]
```
Comment (christulin) [inline src/Components/GameBoard.js:14]: Adding squareActive state, which holds an array of class names.  "square active" signaling the square is currently focused on

File: src/Components/GameBoard.js [hunk 3]
```diff
@@ -1,20 +1,64 @@
...
+        let newNum = num + 3;
+        buttons[newNum].focus();
+        toggleActive[num] = 'square';
+        toggleActive[newNum] = 'square active';
+        this.setState({
+          squareActive: toggleActive
+        })
+      }
+  }
+
```
Comment (christulin) [inline src/Components/GameBoard.js:59]: Added `onKeyDown` event handler that takes two parameters: the index number of square in squareArray and the event object.  Creates copy of `squareActive` array.  Each if/else block will check to see if arrow key was pressed and ensures square is within limit to make change.  3 or 1 added/subtracted from num to identify new square that should be focused on.  swap squareActive with this new array.

Please comment if there are questions here

File: src/Components/GameBoard.js [hunk 4]
```diff
@@ -42,19 +86,19 @@ export class GameBoard extends React.Component {
...
+            <Square value={this.state.squareArray[5]} class={this.state.squareActive[5]} onKeyDown={(e) => this.handleKeyDown(5, e)} onClick={() => this.handleSquareClick(5)} />
           </div>
           <div className="board-row">
-            <Square value={this.state.squareArray[6]} onClick={() => this.handleSquareClick(6)} />
-            <Square value={this.state.squareArray[7]} onClick={() => this.handleSquareClick(7)} />
-            <Square value={this.state.squareArray[8]} onClick={() => this.handleSquareClick(8)} />
+            <Square value={this.state.squareArray[6]} class={this.state.squareActive[6]} onKeyDown={(e) => this.handleKeyDown(6, e)} onClick={() => this.handleSquareClick(6)} />
+            <Square value={this.state.squareArray[7]} class={this.state.squareActive[7]} onKeyDown={(e) => this.handleKeyDown(7, e)} onClick={() => this.handleSquareClick(7)} />
+            <Square value={this.state.squareArray[8]} class={this.state.squareActive[8]} onKeyDown={(e) => this.handleKeyDown(8, e)} onClick={() => this.handleSquareClick(8)} />
```
Comment (christulin) [inline src/Components/GameBoard.js:101]: class of `<Square>` components now references squareActive state.  `OnKeyDown` event added to each instance which calls `handleKeyDown`

File: src/Components/Square.js [hunk 5]
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
 
 const Square = (props) => {
+
     return (
-    <button className="square" onClick={props.onClick}>{props.value}</button>
+    <button type="button" autofocus={props.addFocus} className={props.class} onClick={props.onClick} onKeyDown={props.onKeyDown}>{props.value}</button>
```
Comment (christulin) [inline src/Components/Square.js:6]: will look for prop addFocus and get classname from prop passed to it from Gameboard.js

File: src/Styles/Square.css [hunk 6]
```diff
@@ -2,4 +2,8 @@
     height: 200px;
     width: 200px;
     border: 5px solid black;
+}
+
+.active {
+    border: 5px solid red;
```
Comment (christulin) [inline src/Styles/Square.css:8]: Adding basic red border when class of 'active' added to a square.. we can discuss styling and optimize look/feel in the future

File: src/Components/GameBoard.js [hunk 7]
```diff
@@ -1,20 +1,64 @@
...
+
+      } else if (e.keyCode === 40 && num < 6) {
+        let newNum = num + 3;
+        buttons[newNum].focus();
+        toggleActive[num] = 'square';
+        toggleActive[newNum] = 'square active';
+        this.setState({
+          squareActive: toggleActive
+        })
+      }
```
Comment (debughug) [inline src/Components/GameBoard.js:57]: Extract logic and just condition the newNum to be a lil dryer

File: src/Components/GameBoard.js (hunk 4 above)
Comment (debughug) [inline src/Components/GameBoard.js:101]: Convert the Squares say they use forEach loop and place focus on index 0 and the index value for each squareActive and handleKeyDown

File: src/Components/GameBoard.js (hunk 7 above)
Comment (christulin) [reply src/Components/GameBoard.js]: done

File: src/Components/GameBoard.js (hunk 4 above)
Comment (christulin) [reply src/Components/GameBoard.js]: done

**Pull request merged**